tests/system_metrics/test_system_metrics_logging.py

```diff
@@ -91,3 +91,52 @@ def test_automatic_system_metrics_monitor():
     mlflow.disable_system_metrics_logging()
     mlflow.set_system_metrics_sampling_interval(None)
     mlflow.set_system_metrics_samples_before_logging(None)
+
+
+def test_automatic_system_metrics_monitor_resume_existing_run():
+    mlflow.enable_system_metrics_logging()
+    mlflow.set_system_metrics_sampling_interval(0.2)
+    mlflow.set_system_metrics_samples_before_logging(2)
+    with mlflow.start_run() as run:
+        time.sleep(2)
+
+    # Pause for a bit to allow the system metrics monitoring to exit.
+    time.sleep(1)
+    thread_names = [thread.name for thread in threading.enumerate()]
+    # Check the system metrics monitoring thread has exited.
+    assert "SystemMetricsMonitor" not in thread_names
+
+    # Get the last step.
+    metrics_history = mlflow.tracking.MlflowClient().get_metric_history(
+        run.info.run_id, "system/cpu_utilization_percentage"
+    )
+    last_step = metrics_history[-1].step
+
+    with mlflow.start_run(run.info.run_id) as run:
+        time.sleep(2)
+    mlflow_run = mlflow.get_run(run.info.run_id)
+    metrics = mlflow_run.data.metrics
+
+    expected_metrics_name = [
+        "cpu_utilization_percentage",
+        "system_memory_usage_megabytes",
+        "disk_usage_percentage",
+        "disk_usage_megabytes",
+        "disk_available_megabytes",
+        "network_receive_megabytes",
+        "network_transmit_megabytes",
+    ]
+    expected_metrics_name = [f"system/{name}" for name in expected_metrics_name]
+    for name in expected_metrics_name:
+        assert name in metrics
+
+    # Check the step is correctly resumed.
+    metrics_history = mlflow.tracking.MlflowClient().get_metric_history(
+        run.info.run_id, "system/cpu_utilization_percentage"
+    )
+    assert metrics_history[-1].step > last_step
+
+    # Unset the environment variables to avoid affecting other test cases.
+    mlflow.disable_system_metrics_logging()
+    mlflow.set_system_metrics_sampling_interval(None)
+    mlflow.set_system_metrics_samples_before_logging(None)
```

Comment:
you might want to add this as an autouse fixture in case an error occurs before we reach these lines.

Reply:
nice call.